rubicon/objc/runtime.py

```diff
@@ -909,14 +921,17 @@ def get_ivar(obj, varname):
     ivar = libobjc.class_getInstanceVariable(libobjc.object_getClass(obj), ensure_bytes(varname))
     vartype = ctype_for_encoding(libobjc.ivar_getTypeEncoding(ivar))
 
-    if isinstance(vartype, objc_id):
+    if weak:
+        value = libobjc.objc_loadWeakRetained(obj.value + libobjc.ivar_getOffset(ivar))
+        return libobjc.objc_autoreleaseReturnValue(value)
+    elif isinstance(vartype, objc_id):
         return cast(libobjc.object_getIvar(obj, ivar), vartype)
```

Comment:
When is this cast required? Isn't it sufficient to to return a `objc_id` which later gets converted to the correct `ObjCInstance`?

If it is required, do we need add a similar cast before returning the value in the weak case?

Reply:
You're right that `libobjc.object_getIvar` already returns an `objc_id`. The most common case is that `vartype` is exactly `objc_id`, in which case the cast indeed does nothing. I think it's only needed for the less common case where `vartype` is `Class` (which we have declared as a subclass of `objc_id`) so that the return value is cast to `Class` instead of a plain `objc_id`.

Reply:
Ah, ok! So we should perform a similar cast when returning a value in the weak case as well.